backend/src/core/services/auth.service.ts

```diff
@@ -213,7 +213,7 @@ export const InitAuthService = (redisService: RedisService): AuthService => {
   const sendOtp = async (
     email: string,
     ipAddress: string
-  ): Promise<boolean> => {
+  ): Promise<string | void> => {
```

Comment:
I took a look at the [call site here](https://github.com/opengovsg/postmangovsg/blob/b0729fac7a6a2d2de450846ed0239da15f018f5d/backend/src/core/middlewares/auth.middleware.ts#L36). It seems like it is actually not using the return value at all. If the intention for `sendOtp` is not to use the `string` returned all the time, then As such, the return type should be `Promise<void>`.

As for the rest of the function, I personally think it would be clearer to do `void MailService.mailClient.sendMail({..})` followed by `return`.  


```suggestion
): Promise<void> => {
```

Reply:
then the portion below should be:

```TypeScript
    void MailService.mailClient.sendMail({
      from: config.get('mailFrom'),
      recipients: [email],
      subject: `One-Time Password (OTP) for ${appName}`,
      body: `Your OTP is <b>${otp}</b>. It will expire in ${Math.floor(
        OTP_EXPIRY / 60
      )} minutes.
    Please use this to login to your ${appName} account. <p>If your OTP does not work, please request for a new OTP.</p>
    <p>This login attempt was made from the IP: ${ipAddress}. If you did not attempt to log in to ${appName}, you may choose to investigate this IP address further.</p>
    <p>The ${appName} Support Team</p>`,
    })
    return
```